lib/src/command/upgrade.dart

```diff
@@ -389,6 +391,28 @@ You may have to:
     _showOfflineWarning();
   }
 
+  Pubspec _applyChanges(
```

Comment:
Please give this a documentation comment...

We could also consider tweak `_updatePubspec` to the YAML string instead writing the updated `pubspec.yaml`.

Then we could use `_updatePubspec` and parse YAML again, this has the benefit that:
 * (A) We don't have two methods doing the same thing in different ways.
 * (B) We abort earlier if we fail to parse YAML after modification (or if we fail to modify the YAML).

(B) can happen in certain cases, for example if using references / anchors in YAML in the path being modified.

Reply:
Good idea

Done